### PR TITLE
Feature/issue 105 check box default

### PR DESCRIPTION
**Pull request by SimerdeepSinghGrewal**
added default props to Checkbox component.

### CODE REVIEW

**Review by uparkalau — APPROVED** [review]

**Review by uparkalau — APPROVED** [review]

File: frontend/src/components/Checkbox/Checkbox.jsx [hunk 1]
```diff
@@ -57,7 +79,15 @@ const CustomCheckbox = ({
                   checked={childChecked[index]}
                   onChange={handleChildChange(index)}
                   size={size}
-                  className={`checkbox ${variant}`}
+                  sx={{
```
Comment (erenfn) [inline frontend/src/components/Checkbox/Checkbox.jsx:82]: Can you make this style a variable?

File: frontend/src/components/Checkbox/Checkbox.jsx [hunk 2]
```diff
@@ -1,33 +1,46 @@
...
+  onChange = () => {},
+  variant = 'primary',
+  className = '',
+  style = {},
+  size = 'medium',
+  indeterminate = false,
+  childrenCheckboxes = []
 }) => {
   const [childChecked, setChildChecked] = useState(childrenCheckboxes.map(() => false));
+  const [isChecked, setIsChecked] = useState(checked);
```
Comment (erenfn) [inline frontend/src/components/Checkbox/Checkbox.jsx:28]: something like:

  const  sx = {
    color: variant === 'primary' ? 'var(--main-purple)' : '#808080E5',
    '&.Mui-checked': {
      color: variant === 'primary' ? 'var(--main-purple)' : '#808080E5',
    },
    '&.MuiCheckbox-indeterminate': {
      color: variant === 'primary' ? 'var(--main-purple)' : '#808080E5',
    },
  }

File: frontend/src/components/Checkbox/Checkbox.jsx [hunk 3]
```diff
@@ -37,10 +50,19 @@ const CustomCheckbox = ({
       <FormControlLabel
         control={
           <Checkbox
-            checked={childrenCheckboxes.length > 0 ? childChecked.every(Boolean) : checked}
-            onChange={childrenCheckboxes.length > 0 ? handleParentChange : onChange}
+            checked={childrenCheckboxes.length > 0 ? childChecked.every(Boolean) : isChecked}
+            onChange={childrenCheckboxes.length > 0 ? handleParentChange : handleChange}
             indeterminate={indeterminate || isIndeterminate}
             size={size}
+            sx={{
```
Comment (erenfn) [inline frontend/src/components/Checkbox/Checkbox.jsx:57]: then sx={sx}

**Review by erenfn — CHANGES_REQUESTED** [review]
Also I think you forgot to create a new branch. Can you create another PR without the changes from the tours PR?

**Comment by SimerdeepSinghGrewal** [comment]
> Also I think you forgot to create a new branch. Can you create another PR without the changes from the tours PR?

I did created a new branch "feature/Issue-105-CheckBox-default"

File: frontend/src/components/Checkbox/Checkbox.jsx (hunk 1 above)
Comment (SimerdeepSinghGrewal) [reply frontend/src/components/Checkbox/Checkbox.jsx]: created a function as it needs to change style for each variant

**Comment by SimerdeepSinghGrewal** [comment]
> It still has the changes from the other branch. Please remove those.

done

**Review by uparkalau — DISMISSED** [review]

File: frontend/src/components/ConfirmationPopup/ConfirmationPopup.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
```
Comment (erenfn) [inline frontend/src/components/ConfirmationPopup/ConfirmationPopup.jsx:1]: remove this file

File: frontend/src/App.jsx [hunk 5]
```diff
@@ -23,6 +24,7 @@ function App() {
         <Route path="/check-email" element={<CheckYourEmailPage />} />
         <Route path="/set-new-password" element={<SetNewPasswordPage />} />
         <Route path="/progress-steps" element={<ProgressStepsMain />} />
+        <Route path="/tours" element={<ProductTour />} />
```
Comment (erenfn) [inline frontend/src/App.jsx:27]: remove this line (27)

File: frontend/src/App.jsx [hunk 6]
```diff
@@ -9,6 +9,7 @@ import SetNewPasswordPage from "./scenes/login/SetNewPassword";
 import Private from "./components/Private";
 import { useAuth } from "./services/authProvider";
 import ProgressStepsMain from "./scenes/progressSteps/ProgressStepsMain";
+import ProductTour from "./scenes/productTour/ProductTour";
```
Comment (erenfn) [inline frontend/src/App.jsx:12]: remove this line (12)

**Review by erenfn — CHANGES_REQUESTED** [review]

**Comment by uparkalau** [comment]
@coderabbitai review PR

**Review by uparkalau — DISMISSED** [review]

File: frontend/src/components/ConfirmationPopup/ConfirmationPopup.jsx (hunk 4 above)
Comment (SimerdeepSinghGrewal) [reply frontend/src/components/ConfirmationPopup/ConfirmationPopup.jsx:1]: I never worked on this component but removing it

**Review by uparkalau — APPROVED** [review]

File: frontend/src/components/ConfirmationPopup/ConfirmationPopup.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import PropTypes from 'prop-types';
```
Comment (erenfn) [inline frontend/src/components/ConfirmationPopup/ConfirmationPopup.jsx:1]: delete this file

**Review by erenfn — CHANGES_REQUESTED** [review]

File: frontend/src/components/ConfirmationPopup/ConfirmationPopup.jsx (hunk 7 above)
Comment (SimerdeepSinghGrewal) [reply frontend/src/components/ConfirmationPopup/ConfirmationPopup.jsx:1]: I never worked on this component

Comment (erenfn) [reply frontend/src/components/ConfirmationPopup/ConfirmationPopup.jsx]: Not line 1. Delete the entire file

**Review by erenfn — APPROVED** [review]

**Pull request merged**